plugin-server/src/worker/ingestion/person-state.ts

```diff
@@ -158,10 +147,30 @@ export class PersonState {
             }
         }
 
-        // Person was likely created in-between start-of-processing and now, so ensure that subsequent queries
-        // to fetch person still return the right `person`
-        this.personContainer = this.personContainer.reset()
-        return false
+        return [await this.getPersonOrError(), false]
+    }
+
+    private async getPersonOrError(): Promise<Person> {
+        const person = await this.db.fetchPerson(this.teamId, this.distinctId)
+        if (!person) {
+            // There are three options for how we got here:
+            // - person creation is ongoing (but we're currently not parallel processing ingestion for the same ID)
+            // - person creation failed (rare)
+            // - person was deleted (rare)
+            // TODO: retries for creation and/or fetching
+            const error = Error('Race condition: Person has been deleted or creation failed')
+            status.error('🚨', 'get_person_failed', { error, teamId: this.teamId, distinctId: this.distinctId })
+            Sentry.captureException(error, {
+                extra: {
+                    teamId: this.teamId,
+                    distinctId: this.distinctId,
+                    timestamp: this.timestamp,
+                    personUuid: this.newUuid,
+                },
+            })
+            throw error
```

Comment:
sidetrack: we'll need to try out (on dev or local devenv) what happens when PG does down: do we properly retry here, or still send to DLQ? If the later, that can be fixed in a second PR.

Ideally, `createOrGetPerson` should emit the `DependencyUnavailableError`, but we'll also want to retry here if a race happens, right?

Reply:
yep, that's what I thought. We'll improve on another PR, it's not a regression.